File: src/py-opentimelineio/opentimelineio/schema/string_stream_address.py
# SPDX-License-Identifier: Apache-2.0
# Copyright Contributors to the OpenTimelineIO project

from .. core._core_utils import add_method
from .. import _otio


@add_method(_otio.StringStreamAddress)
def __str__(self):
    return "StringStreamAddress({})".format(self.address)


@add_method(_otio.StringStreamAddress)
def __repr__(self):
    return "otio.schema.StringStreamAddress(address={})".format(
        repr(self.address)
    )

File: src/opentimelineio/stringStreamAddress.cpp
// SPDX-License-Identifier: Apache-2.0
// Copyright Contributors to the OpenTimelineIO project

#include "opentimelineio/stringStreamAddress.h"

namespace opentimelineio { namespace OPENTIMELINEIO_VERSION_NS {

StringStreamAddress::StringStreamAddress(std::string const& address)
    : Parent()
    , _address(address)
{}

StringStreamAddress::~StringStreamAddress()
{}

bool
StringStreamAddress::read_from(Reader& reader)
{
    return reader.read("address", &_address) && Parent::read_from(reader);
}

void
StringStreamAddress::write_to(Writer& writer) const
{
    Parent::write_to(writer);
    writer.write("address", _address);
}

}} // namespace opentimelineio::OPENTIMELINEIO_VERSION_NS
